apps/files/models.py

```diff
@@ -602,12 +602,13 @@ def save(self, *args, **kw):
         if self.validation:
             if self.load_validation()['errors'] == 0:
                 self.valid = True
-        super(FileUpload, self).save()
+        super(FileUpload, self).save(*args, **kw)
 
     def add_file(self, chunks, filename, size):
-        if not self.pk:
-            self.save()
-        filename = smart_str(u'{0}_{1}'.format(self.pk, filename))
+        force_insert = not self.uuid
+        if force_insert:
+            self.uuid = self._meta.get_field('uuid')._create_uuid().hex
```

Comment:
I updated this to use `UUIDField`'s uuid generation method. There are a couple `_` property accesses but it doesn't force this to be uuid4.